runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java

```diff
@@ -83,9 +83,13 @@
         // to the correct version, reflecting any optimistic
         // updates.
         return proxy.getUnderlyingObject().access(
-                o -> (getWriteSetEntryList(proxy.getStreamID()) == null && // No updates
+                o -> (getWriteSetEntryList(proxy.getStreamID()).size() == 0 && // No updates
```

Comment:
actually, getWriteSetEntryList returns empty collection, not null, on empty write set. This is the main reason for introducing this fix, we actually -always- take a write lock on every access.

I don't mind addding this check, to prevent exception occurring if getWriteSetEntryList changes in the future.